src/monstergenerator.cpp

```diff
@@ -786,6 +786,14 @@ void mtype::load( const JsonObject &jo, const std::string &src )
     optional( jo, was_loaded, "zombify_into", zombify_into, auto_flags_reader<mtype_id> {},
               mtype_id() );
 
+    // Disable upgrading when JSON contains `"fungalized": false`, but fallback to the
+    // normal behavior if "fungalized" is not boolean or not `false`.
+    if( jo.has_bool( "fungalized" ) && !jo.get_bool( "fungalized" ) ) {
+        fungalized = mtype_id();
+    } else if( jo.has_member( "fungalized" ) ) {
+        fungalized = mtype_id( jo.get_string( "fungalized" ) );
+    }
```

Comment:
You should reuse the same implementation as `zombify_into`, it will be easier to implement, look more coherent for people wanting to add fungus version and using `optional` will make it moddable. Maybe rename it `fungalize_into` to make it really clear what it does.